packages/@aws-cdk/cdk/lib/physical-name.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * Represents the human-readable, physical name of a resource.
+ * These can be known at code runtime,
+ * in the case of fixed, customer-provided names,
+ * at synthesis time, in case of automatically assigned names by the framework,
+ * or only at deploy time, assigned by CloudFormation -
+ * which is the default.
+ *
+ * @see #deployTime()
+ * @see #deployTimeOrAssigned()
+ * @see #fixed()
+ */
+export abstract class PhysicalName {
+  /**
+   * A physical name that will be automatically assigned at deploy time.
+   * This is the default you don't specify a physical name.
+   */
+  public static deployTime(): PhysicalName {
+    return new DeployTimePhysicalName();
+  }
+
+  /**
+   * A physical name that will be assigned at synthesis time if the resource
+   * is used in a cross-environment manner,
+   * and at deploy time otherwise.
+   * This is the default name implicitly created sub-resources
+   * (like IAM Roles) are created with in the Construct Library.
+   * If you're using a Construct in a cross-environment manner,
+   * you need to use either this name,
+   * or a {@link #fixed() fixed name}.
+   */
+  public static deployTimeOrAssigned(): PhysicalName {
+    return new DeployTimeOrAssignedPhysicalName();
+  }
+
+  /**
+   * A fixed physical name (one that is known statically, at synthesis time).
+   *
+   * @param name the name to assign
+   */
+  public static fixed(name: string): PhysicalName {
```

Comment:
Let's just of `PhysicalName.of` instead of `PhysicalName.fixed`. To me, "fixed" indicates that the name cannot be changed after it's been defined, which is actually true for both auto-generated and names provided by the user, so I think something like `of` would read better:

```ts
new Bucket(this, 'bucket', { bucketName: PhysicalName.of('my-bucket') });
```

Reply:
Done.